tsk/manager.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+set -v
+
+CANARY="/persist/alreadyRW"
+
+cleanup() {
+  if [ -f ${CANARY} ]; then
+    # /persist was already RW before we started.
+    # Remove the marker and don't remount as read-only.
+    rm ${CANARY}
+  else
+    # Remount /persist as read-only
+    sudo mount -o remount,ro /persist
+  fi
```

Comment:
This is smart, however, because we know `/persist` shall never be open really, I'd say keep the `cleanup` simple and always remount as `ro` even if it was `rw` prior. I am paranoid on that partition, as the risk is significant.

Reply:
The intent was to make this PR cherrypick-able by all forks. Some forks like FP likes to write in /persist so I didn't want to disrupt that. If it's a blocker I can make a commit that changes the behavior, which some forks maintainer may choose to skip

Reply:
In fact, thinking about it, we don't really want to be the ones that open this partition. So for us we'd really not allow this at all, unless we absolutely must. 

Since SP will be protecting the key via sunnylink, in our case it would be more than enough to have it on the params. What we can do is try to read from /persist in hopes of another fork maybe had written it in the past, but that'd be it. And alternatively we'd allow writing it to `/data/media/0/tsk` which should survive agnos updates (but not resets afaik)

Reply:
Okay I'll make a version that doesn't write to `/persist`. It will still try to read from it, in case a different fork allowed writing to `/persist` and archived the key there

Reply:
No longer applicable